app/src/main/java/com/keylesspalace/tusky/components/compose/ImageDownsizer.kt

```diff
@@ -0,0 +1,113 @@
+/* Copyright 2022 Tusky contributors
+ *
+ * This file is a part of Tusky.
+ *
+ * This program is free software; you can redistribute it and/or modify it under the terms of the
+ * GNU General Public License as published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Tusky is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even
+ * the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+ * Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License along with Tusky; if not,
+ * see <http://www.gnu.org/licenses>. */
+
+package com.keylesspalace.tusky.components.compose
+
+import android.content.ContentResolver
+import android.graphics.Bitmap
+import android.graphics.Bitmap.CompressFormat
+import android.graphics.BitmapFactory
+import android.net.Uri
+import com.keylesspalace.tusky.util.IOUtils
+import com.keylesspalace.tusky.util.calculateInSampleSize
+import com.keylesspalace.tusky.util.getImageOrientation
+import com.keylesspalace.tusky.util.reorientBitmap
+import java.io.File
+import java.io.FileNotFoundException
+import java.io.FileOutputStream
+import java.io.InputStream
+import java.io.OutputStream
+
+/**
+ * @param uri             the uri pointing to the input file
+ * @param sizeLimit       the maximum number of bytes the output image is allowed to have
+ * @param contentResolver to resolve the specified input uri
+ * @param tempFile        the file where the result will be stored
+ * @return true when the image was successfully resized, false otherwise
+ */
+fun downsizeImage(
+    uri: Uri,
+    sizeLimit: Int,
+    contentResolver: ContentResolver,
+    tempFile: File
+): Boolean {
+
+    var inputStream: InputStream?
+    inputStream = try {
```

Comment:
why do we use the same variable twice?
Can we use `with` or `use`?

Reply:
Hmm, with use/with it gets a bit complicated since there are three streams involved, I'll leave it like it is